cluster/gce/gci/configure.sh

```diff
@@ -549,30 +549,33 @@ function install-containerd-ubuntu {
 }
 
 function install-auth-provider-gcp {
-  local -r auth_provider_tar="auth-provider-gcp-${DEFAULT_AUTH_PROVIDER_GCP_VERSION}-${HOST_PLATFORM}_${HOST_ARCH}.tar.gz"
-  echo "Downloading auth-provider-gcp ${auth_provider_tar}" .
+  local -r filename="auth-provider-gcp"
+  local -r auth_provider_storage_full_path="${AUTH_PROVIDER_GCP_STORAGE_PATH}/${AUTH_PROVIDER_GCP_VERSION}/${HOST_PLATFORM}_${HOST_ARCH}/${filename}"
+  echo "Downloading auth-provider-gcp ${auth_provider_storage_full_path}" .
 
-  local -r auth_provider_release_path="https://storage.googleapis.com/cloud-provider-gcp"
-  download-or-bust "${DEFAULT_AUTH_PROVIDER_GCP_HASH_AMD64}" "${auth_provider_release_path}/${auth_provider_tar}"
+  case "${HOST_ARCH}" in
+    amd64)
+      local -r auth_provider_gcp_hash="${AUTH_PROVIDER_GCP_HASH_LINUX_AMD64}"
+      ;;
+    arm64)
+      local -r auth_provider_gcp_hash="${AUTH_PROVIDER_GCP_HASH_LINUX_ARM64}"
+      ;;
+    *)
+      echo "Unrecognized version and platform/arch combination: ${HOST_PLATFORM}/${HOST_ARCH}"
+      exit 1
+  esac
 
-  # Keep in sync with --image-credential-provider-bin-dir in ../util.sh
-  local auth_provider_dir="${KUBE_HOME}/auth-provider-gcp"
-  mkdir -p "${auth_provider_dir}"
-  tar xzf "${KUBE_HOME}/${auth_provider_tar}" -C "${auth_provider_dir}" --overwrite
-  mv "${auth_provider_dir}/auth-provider-gcp" "${KUBE_BIN}"
-  chmod a+x "${KUBE_BIN}/auth-provider-gcp"
+  download-or-bust "${auth_provider_gcp_hash}" "${auth_provider_storage_full_path}"
 
-  rm -f "${KUBE_HOME}/${auth_provider_tar}"
-  rmdir "${auth_provider_dir}"
+  mv "${KUBE_HOME}/${filename}" "${AUTH_PROVIDER_GCP_LINUX_BIN_DIR}"
+  chmod a+x "${KUBE_BIN}/${filename}"
```

Comment:
@dims @pawbana this looks like a bug, unless AUTH_PROVIDER_GCP_LINUX_BIN_DIR == KUBE_BIN

Reply:
Yes this looks like a bug.  I'll prepare fix for it.